centreon/packages/ui/src/Graph/Graph.tsx

```diff
@@ -0,0 +1,273 @@
+            width={graphWidth}
+            xScale={xScale}
+            {...grids}
+          />
+
+          <Axes
+            data={{
+              baseAxis,
+              lines,
+              timeSeries,
+              ...axis
+            }}
+            graphEndpoint={endpoint}
+            height={graphHeight}
+            leftScale={leftScale}
+            rightScale={rightScale}
+            width={graphWidth}
+            xScale={xScale}
+          />
+
+          <Lines
+            anchorPoint={{
+              renderRegularLinesAnchorPoint: ({
+                areaColor,
+                lineColor,
+                metric,
+                timeSeries: regularLinesTimeSeries,
+                transparency,
+                xScale: xScaleRegularLines,
+                yScale
+              }: RegularLinesAnchorPoint): JSX.Element => (
+                <g>
+                  {displayRegularLinesAnchorPoint && (
+                    <RegularAnchorPoint
+                      areaColor={areaColor}
+                      lineColor={lineColor}
+                      metric={metric}
+                      timeSeries={regularLinesTimeSeries}
+                      transparency={transparency}
+                      xScale={xScaleRegularLines}
+                      yScale={yScale}
+                      {...commonAnchorPoint}
+                      {...anchorPoint?.areaRegularLinesAnchorPoint}
+                    />
+                  )}
+                </g>
+              ),
+              renderStackedAnchorPoint: ({
+                areaColor,
+                transparency,
+                lineColor,
+                stack,
+                xScale: x,
+                yScale: y
+              }: StackedAnchorPointModel): JSX.Element => (
+                <g>
+                  {displayStackedAnchorPoint && (
+                    <StackedAnchorPoint
+                      areaColor={areaColor}
+                      lineColor={lineColor}
+                      stackValues={stack as unknown as Array<StackValue>}
+                      transparency={transparency}
+                      xScale={x}
+                      yScale={y}
+                      {...commonAnchorPoint}
+                      {...anchorPoint?.areaStackedLinesAnchorPoint}
+                    />
+                  )}
+                </g>
+              )
+            }}
+            height={graphHeight}
+            shape={{ areaRegularLines, areaStackedLines }}
+          />
+
+          <InteractionWithGraph
+            renderAreaToInteractWith={
+              <Bar
+                className={classes.overlay}
+                fill="transparent"
+                height={graphHeight}
+                width={graphWidth}
+                x={0}
+                y={0}
+                onMouseDown={mouseDown}
+                onMouseLeave={mouseLeave}
+                onMouseMove={mouseMove}
+                onMouseUp={mouseUp}
+              />
+            }
```

Comment:
Instead of rendering the Bar itself. I think it's better that the `InteractionWithGraph` component handles mouse events and use atoms